### ISSUE TITLE
Showing whether a package has a prefix

### ISSUE THREAD

**Issue by jakirkham** [Anaconda-Platform/support#44]
Is there a way to determine whether a package on anaconda.org has a prefix? Namely showing through the web whether it has the file `has_prefix` in `info`. This would really help us quickly determine the scope of things needing to be rebuilt with `conda-build` 2.0.0.

cc @msarahan

**Comment by stephenakearns** [comment]
Hi @jakirkham - can you provide an example of a package that is using `has_prefix`

For Reference: from http://conda.pydata.org/docs/spec.html ... 

> info/has_prefix: This optional file lists all files that contain a hard-coded build prefix or placeholder prefix, which needs to be replaced by the install prefix at installation time. Note that due to the way the binary replacem

ent works, the placeholder prefix must be longer than the install prefix.

**Comment by stephenakearns** [comment]
@jakirkham - checking to see a proposed location for this info makes sense from a user perspective: on the files tab, when clicking the little "info" icon on a given line, and displaying the has_prefix info in the table info that appear sin the popup

![image](https://cloud.githubusercontent.com/assets/13372164/16126412/a693f858-33bd-11e6-9d3a-0282454ab950.png)

**Comment by jakirkham** [comment]
Sure, a simple but common one would be `pkg-config`. Take a look at any Mac or Linux package [here]( https://anaconda.org/conda-forge/pkg-config/files ). I have verified that build number 1 has this. If you need more examples, take a look at [`curl`]( https://anaconda.org/conda-forge/curl/files ), [`git`]( https://anaconda.org/conda-forge/git/files ), or [`icu`]( https://anaconda.org/conda-forge/icu/files ).

Sure that location looks reasonable.

It would also be nice if we can access this info programmatically. Being able to compile a list quickly is pretty helpful in this process.

**Comment by stephenakearns** [comment]
Ok - let's implement the UI part, @jakirkham can you create a separate issue to implement in the API?

**Comment by jakirkham** [comment]
Sure I cobbled together an issue ( https://github.com/Anaconda-Platform/support/issues/48 ) for this. May need some more tweaking/rewording, but it has the basic idea.

**Comment by jakirkham** [comment]
Thanks again for looking into this. Being able to determine what we need to rebuild is going to be a huge help for this transition.

**Comment by bollwyvl** [comment]
PR up to add this introspection to `anaconda-client`: Anaconda-Platform/anaconda-client#342 

Once that lands/gets shipped (that process is a little opaque to me) it will "just work" and start showing up (if true) in the UI and the API. Turns out this will also take care of #48 as well.

Back-filling the existing data is a whole other ballgame...

**Comment by bollwyvl** [comment]
here's what one with has_prefix will look like:
```yaml
# /api/release/vagrant/curl/7.49.1

{
  "distributions": [
    {
      "attrs": {
        "arch": "x86_64", 
        "build": "0", 
        "build_number": 0, 
        "depends": [
          "openssl", 
          "zlib"
        ], 
        "has_prefix": true, # hooray!
        "license": "MIT/X derivate (http://curl.haxx.se/docs/copyright.html)", 
        "license_family": "MIT", 
        "machine": "x86_64", 
        "operatingsystem": "darwin", 
        "platform": "osx", 
        "subdir": "osx-64", 
        "target-triplet": "x86_64-any-darwin"
      }, 
```
... and without...
```yaml
# /api/release/vagrant/curl/7.48.0
{
  "distributions": [
    {
      "attrs": {
        "arch": "x86_64", 
        "build": "vc14_2", 
        "build_number": 2, 
        "depends": [
          "zlib"
        ], 
        "features": "vc14", 
        "has_prefix": false, # also hooray!
        "license": "MIT/X derivate (http://curl.haxx.se/docs/copyright.html)", 
        "license_family": "MIT", 
        "machine": "x86_64", 
        "operatingsystem": "win32", 
        "platform": "win", 
        "subdir": "win-64", 
        "target-triplet": "x86_64-any-win32"
      }, 
```

A legacy package will lack `has_prefix` key.

**Comment by jakirkham** [comment]
This is life changing, @bollwyvl. My fear of `conda-build` version 2.0.0 has gone down significantly since seeing this addition.

**Comment by bollwyvl** [comment]
@jakirkham Glad to be of service. 🎩👌 As i mentioned, getting all the data backfilled will require coordinating with more people!

**Comment by jakirkham** [comment]
> getting all the data backfilled will require coordinating with more people

Yep, noted. Though this already gets us much closer.

**Comment by bollwyvl** [comment]
@jakirkham This has been released with [`anaconda-client` 1.5.1](https://github.com/Anaconda-Platform/anaconda-client/blob/1.5.1/CHANGELOG.md#151-2016-08-01)!

Here is potentially the very first package uploaded with `has_prefix` populated:
https://api.anaconda.org/release/nbcio/git/2.8.2

(should have picked a smaller file!)

Let us know what else we can do to make the transition to `conda-build` 2.0 easier for your needs!

**Comment by msarahan** [comment]
Thanks @bollwyvl

**Comment by jakirkham** [comment]
Thanks @bollwyvl. We have been using 1.5.1 successfully for the past few days. Here's hoping we can start keeping better records of this info going forward.

**Comment by jakirkham** [comment]
Have opened a new issue ( https://github.com/Anaconda-Platform/support/issues/57 ) to handle all the packages that were uploaded before `anaconda-client` version `1.5.1`.
